scripts/git-purge-files

```diff
@@ -1,46 +1,142 @@
 #!/usr/bin/perl
 #
-# Purge files from Git repositories.
-#
-# Attention:
-#     You want to run this script on a case sensitive file-system (e.g.
-#     ext4 on Linux). Otherwise the resulting Git repository will not
-#     contain changes that modify the casing of file paths.
-#
-# Usage:
-#     git-purge-files [path-regex1] [path-regex2] ...
-#
-# Examples:
-#    Remove the file "test.bin" from all directories:
-#    git-purge-path "/test.bin$"
-#
-#    Remove all "*.bin" files from all directories:
-#    git-purge-path "\.bin$"
-#
-#    Remove all files in the "/foo" directory:
-#    git-purge-path "^/foo/$"
-#
-# Author: Lars Schneider, https://github.com/larsxschneider
+# Purge files from Git repositories
 #
 
+use 5.010;
 use strict;
 use warnings;
+use Getopt::Std;
+use File::Temp qw/ tempdir /;
+
+sub usage() {
+    print STDERR <<END;
+NAME
+       git-purge-files - Purge files from Git repositories
+
+SYNOPSIS
+       git-purge-files [-c] [-d] [-h] [<path-regex>] ...
+
+
+DESCRIPTION
+       This command purges files from a Git history by rewriting all
+       commits. Please note that this changes all commit hashes in the
+       history and therefore all branches and tags.
+
+       You want to run this script on a case sensitive file-system (e.g.
+       ext4 on Linux). Otherwise the resulting Git repository will not
+       contain changes that modify the casing of file paths.
+
+OPTIONS
+       <path-regex>...
+           A list of regular expression that defines what files should
+           be purged from the history. Use a `/` to anchor a path to the
+           root of the repository.
+
+       -c
+           Run in checking mode. The script will run the underlaying
+           `git fast-export | git fast-import` command without any
+           modifications to the data stream. Afterwards the input
+           repository is compared against the output repository.
+
+           ATTENTION: Although we run a check here, the repository
+                      under test is rewritten and potentially modified!
+
```

Comment:
@larsxschneider I think it would be good to add _why_ you want to run the checking mode? I assume you would want to run it to check if you can use the faster `-d` mode?
Maybe add something to `DESCRIPTION` saying:
> For large repositories we recommend to run this script in checking mode (`-c`) mode first in order to determine if it can run in the much faster diff mode (`-d`) mode.

?

Reply:
Great idea 👍